style/layer/road.js

```diff
@@ -0,0 +1,367 @@
+//At this zoom, render switches from unified to differentiated bridge/tunnel rendering
```

Comment:
Get in the habit of putting

```js
"use strict";
```

at the top of js files. (Or switch to ES modules / typescript).

Reply:
Just added this to all of the project's js files.  I wasn't sure what "switch to ES modules / typescript" means or which approach is preferred.

Reply:
Note that the bodies of classes are always in strict mode, but this directive enables it for the file as a whole. https://mdn.io/Strict_mode

As briefly as possible:

At the moment, the modules are all linked together via "bunch of script tags", and the dependencies are all implicitly via global variables; writing any sort of tests would be somewhat cumbersome as there's nothing to import as such.

So using some sort of module system is desirable. https://mdn.io/Modules

The main issue is javascript's historical lack of any module system. So 'CommonJS' was developed in the nodejs ecosystem, and build systems exist to port all of these `require()` calls into something that the browser can execute, and also combining code into one big file, and eliminating dead code, etc. Various compilers and compiler accessories exist (babel, gulp, grunt, rollup, webpack, etc.) but two worth mentioning are:

- TypeScript: Adds useful things at the compile step by adding static type annotations, has serious traction as a typed layer on top of javascript.
- ESBuild: Modern and very fast. Some of these compilers have been notoriously slow.

And now, javascript supports real modules (with `import`), and they can be natively supported by browsers without a build step. Which is cool. But a build step is still useful to have in order to get bundling, dead code elimination, type checking, etc. And you can get that with TypeScript or ESBuild as well. And your build target can still be ES modules (for modern browsers and toolchains) or commonjs for historical reasons.

So, if your source files are ES modules, then they're always in ES strict mode. And you really always want to be in strict mode. So, add the directive, or switch your source files to ES modules.

Reply:
I'm all for switching to ES modules 👍.  Would be nice to not have everything in the global namespace.

Reply:
Oh, I was wondering if there was a better way to deal with the big pile of script tags.  That _does_ seem a lot easier.

Reply:
Okay, I've made the switch to ES modules.  Way cleaner now!